Source/Particles/RigidInjectedParticleContainer.cpp

```diff
@@ -77,12 +77,37 @@ void RigidInjectedParticleContainer::InitData()
 
     AddParticles(0); // Note - add on level 0
 
+    // Store velocity with which rigid particles move
+    vzbeam_ave_boosted = meanParticleVelocity(false)[2];
+    
     // Particles added by AddParticles should already be in the boosted frame
     RemapParticles();
 
     Redistribute();  // We then redistribute
 }
 
+void
+RigidInjectedParticleContainer::PostRestart()
+{
+    // Call parent class
+    PhysicalParticleContainer::PostRestart();
+
+    // Store velocity with which rigid particles move
+    vzbeam_ave_boosted = meanParticleVelocity(false)[2];
```

Comment:
I don't think that this is correct in general. If the checkpoint was done when part of the beam has been injected, this won't give the same result since the injected particles have been advanced. I think that `vzbeam_ave_boosted` needs to be saved in the checkpoint.

Reply:
@RemiLehe similar to https://github.com/ECP-WarpX/WarpX/pull/2058/files

But for particle container specific attributes, see this logic:
`WarpXParticleContainer::ReadHeader` & `WarpXParticleContainer::WriteHeader`

You might want to add a `GotoNextLine(is);` after charge & mass in those, though.
Discussed: maybe also write a dummy value as fallback for non-rigid injected particles and read in also back to something you discard.